### PR TITLE
QoL change: List file sizes atop names

### PR DESCRIPTION
**Pull request by Virepri**
Tackles issue #317.

This solution isn't *perfect* but it more than gets the job done. Would it be preferable if an optional parameter was added to treat 1 GB as 1000MB as some OSes, hard drive manufacturers, etc. do?

Atop that, would it be preferable to enable the ability to keep a running tally on file size and file count?

### CODE REVIEW

File: cmd/list.go [hunk 1]
```diff
@@ -163,3 +165,25 @@ func printListContainerResponse(lsResponse *common.ListContainerResponse) {
 		glcm.Info(lsResponse.Blobs[index])
 	}
 }
+
+func ByteSizeToString(size int) string {
```
Comment (zezha-msft) [inline cmd/list.go]: we can make it lower-case since it doesn't need to be exported

File: cmd/list.go [hunk 2]
```diff
@@ -163,3 +165,25 @@ func printListContainerResponse(lsResponse *common.ListContainerResponse) {
...
+	unit := 0
+	floatSize := float64(size)
+	gigSize := 1024
+
+	for floatSize/float64(gigSize) >= 1 {
+		unit++
+		floatSize /= float64(gigSize)
+	}
+
+	return strconv.FormatFloat(floatSize, 'f', 2, 64) + " " + units[unit]
```
Comment (zezha-msft) [inline cmd/list.go]: please consider adding some unit tests for validation

File: cmd/list.go [hunk 3]
```diff
@@ -163,3 +165,25 @@ func printListContainerResponse(lsResponse *common.ListContainerResponse) {
 		glcm.Info(lsResponse.Blobs[index])
 	}
 }
+
+func ByteSizeToString(size int) string {
+	units := []string{
+		"bytes",
```
Comment (zezha-msft) [inline cmd/list.go]: abbreviations would read better, what do you think?

File: cmd/list.go (hunk 1 above)
Comment (Virepri) [reply cmd/list.go]: No longer exported as of [9cc9d04](https://github.com/Azure/azure-storage-azcopy/pull/341/commits/9cc9d04ee8a8bc98aee541a17413c42e0f359cfe)

File: cmd/list.go (hunk 3 above)
Comment (Virepri) [reply cmd/list.go]: Abbreviations inserted in [9cc9d04](https://github.com/Azure/azure-storage-azcopy/pull/341/commits/9cc9d04ee8a8bc98aee541a17413c42e0f359cfe)

File: cmd/list.go (hunk 2 above)
Comment (Virepri) [reply cmd/list.go]: Unit tests added in [9cc9d04](https://github.com/Azure/azure-storage-azcopy/pull/341/commits/9cc9d04ee8a8bc98aee541a17413c42e0f359cfe)

File: cmd/bytesizetostring_test.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
+package cmd
+
+import (
+	"testing"
+)
+
+func TestBToString(t *testing.T) {
```
Comment (zezha-msft) [inline cmd/bytesizetostring_test.go]: Thanks for the tests! Please consider using the `check` package for validation. Refer to other test files in the cmd module.

**Comment by seguler** [comment]
Wohoo! 

Some feedback: Can we return sizes in bytes by default so it is easy to do sum using PowerShell or bash ?

There could still be an option such as --human-readable to show bytes with KB/MB/GB units.

Thoughts ?

**Comment by seguler** [comment]
Also any chance we can add a counter that we could print total size at the end of the listing ?

**Comment by Virepri** [comment]
Absolutely. I added those as potential changes I'd be willing to make in the PR comment in the first place. I'll get to it once I find a good time.

**Comment by seguler** [comment]
thanks @Virepri !
@zezha-msft do you agree with my proposal ?

**Comment by zezha-msft** [comment]
@seguler :shipit: actually I think a `machine-readable` flag might be better. In other words we display human readable sizes by default, and avoid doing that if the flag is true.

**Comment by zezha-msft** [comment]
@Virepri you should have write access now, please push to a branch on the repo to trigger CI.

**Comment by Virepri** [comment]
@seguler's suggestion has been added, as well as feedback from @JohnRusk has been taken.

I think this is about as good as it's gonna get

**Comment by zezha-msft** [comment]
tracking with #342

**Pull request closed without merging**